pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -201,7 +201,10 @@
     // Specify the tls cipher the broker will use to negotiate during TLS Handshake.
     // Example:- [TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256]
     private Set<String> tlsCiphers = Sets.newTreeSet();
-    
+    // Specify whether Client certificates are required for TLS
+    // https://netty.io/4.0/api/io/netty/handler/ssl/ClientAuth.html
+    // Acceptable values:- [NONE, OPTIONAL, REQUIRE]
+    private String tlsClientAuth = "OPTIONAL";
```

Comment:
We should try to use an enum, instead of string

Reply:
Can't use enums, using enums gives an error 
```
java.lang.IllegalArgumentException: failed to initialize tlsClientAuth field while setting value ClientAuth.OPTIONAL
	at org.apache.pulsar.common.util.FieldParser.lambda$update$0(FieldParser.java:126) ~[pulsar-common.jar:2.0.0-incubating-SNAPSHOT]
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:1.8.0_121]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:580) ~[?:1.8.0_121]
	at org.apache.pulsar.common.util.FieldParser.update(FieldParser.java:120) ~[pulsar-common.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.common.configuration.PulsarConfigurationLoader.create(PulsarConfigurationLoader.java:89) ~[pulsar-broker-common.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.common.configuration.PulsarConfigurationLoader.create(PulsarConfigurationLoader.java:73) ~[pulsar-broker-common.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.PulsarBrokerStarter.loadConfig(PulsarBrokerStarter.java:55) ~[pulsar-broker.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.PulsarBrokerStarter.access$0(PulsarBrokerStarter.java:52) ~[pulsar-broker.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.PulsarBrokerStarter$BrokerStarter.<init>(PulsarBrokerStarter.java:124) ~[pulsar-broker.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.PulsarBrokerStarter.main(PulsarBrokerStarter.java:233) ~[pulsar-broker.jar:2.0.0-incubating-SNAPSHOT]
Caused by: java.lang.UnsupportedOperationException: Cannot convert from java.lang.String to io.netty.handler.ssl.ClientAuth. Requested converter does not exist.
	at org.apache.pulsar.common.util.FieldParser.convert(FieldParser.java:95) ~[pulsar-common.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.common.util.FieldParser.value(FieldParser.java:157) ~[pulsar-common.jar:2.0.0-incubating-SNAPSHOT]
	at org.apache.pulsar.common.util.FieldParser.lambda$update$0(FieldParser.java:124) ~[pulsar-common.jar:2.0.0-incubating-SNAPSHOT]
	... 9 more
```

Reply:
I was meaning to have `FieldParser` to handle enums. 
At the very least, we should initialize here the string to `ClientAuth.OPTIONAL.toString()` and take an enum in `setTlsClientAuth()`